server/middleware/patchable.coffee

```diff
@@ -54,3 +63,97 @@ module.exports =
       watchers = _.filter watchers, (id) -> not id.equals(req.user._id)
       doc.set('watchers', watchers)
     res.status(200).send(doc)
+
+    
+  postPatch: (Model, collectionName, options={}) -> wrap (req, res) ->
+    # handle either "POST /db/<collection>/:handle/patch" or "POST /db/patch" with target included in body
+    if req.body.target?.id
+      target = yield Model.findById(req.body.target.id)
+    else
+      target = yield database.getDocFromHandle(req, Model)
+    if not target
+      throw new errors.NotFound('Target not found.')
+
+    # normalize the delta because in tests, changes to patches would sneak in and cause false positives
+    # TODO: Figure out a better system. Perhaps submit a series of paths? I18N Edit Views already use them for their rows.
+    originalDelta = req.body.delta
+    originalTarget = target.toObject()
+    newTargetAttrs = _.cloneDeep(target.toObject(), (value) ->
+      return value if value instanceof mongoose.Types.ObjectId
+      return value if value instanceof Date
+      return undefined
+    )
```

Comment:
_.cloneDeep can't handle class instances, it seems. Weird things happen to ObjectIds at least when they're "cloned". I did some testing on Dates, and I *think* they're fine... but I'm going to leave the logic here just in case.